### DISCUSSION TITLE
Vlocode error with retrieve and deploy v0.17.12

### DISCUSSION THREAD

**Discussion by luisphp** [Codeneos/vlocode#391] [General]
Hello i'm trying to retrieve and deploy datapacks using vlocde version v0.17.12 and always get this error message, any seggestion? 

results.find is not a function

![image](https://user-images.githubusercontent.com/17937238/201152351-bf53d562-d7e2-4431-ab70-5168be3b9b25.png)

thanks!

**Comment by Codeneos** [comment]
Not sure, I haven't see this before.

The only place where I know `results.find` is used is in [salesforceService.getInstalledPackageDetails()](https://github.com/Codeneos/vlocode/blob/8c5bfaf6755358275997376c9d83ee169be10986/packages/salesforce/src/salesforceService.ts#L107)

But the function that underlies it should always return an array (metadata API call to list installed packages).

**Comment by Codeneos** [comment]
I did some digging and turns out it might be that function that I quoted earlier.

Attached an updated version of the extension that has a fix for [salesforceService.getInstalledPackageDetails()](https://github.com/Codeneos/vlocode/blob/8c5bfaf6755358275997376c9d83ee169be10986/packages/salesforce/src/salesforceService.ts#L107) which ensures it always returns an array.

Let me know if it fixes the above; if so I'll create an intermedia release that fixes the issue :)

[vlocode-0.17.12.vsix.zip](https://github.com/Codeneos/vlocode/files/9993182/vlocode-0.17.12.vsix.zip)

To install the extension unzip `vsix` file and manually install it from the extension tab "Install from VSIX...", see the screenshot below:

![image](https://user-images.githubusercontent.com/787686/201430191-51d9e3d6-1a1f-43ec-a4d1-18ab72b64937.png)
